src/main/java/org/sagebase/redirect/filter/RedirectFilter.java

```diff
@@ -33,7 +33,7 @@
 		"<!DOCTYPE html PUBLIC\"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n" +
 		"<html xmlns=\"http://www.w3.org/1999/xhtml\">\n" +
 		"<script type=\"text/javascript\">\n" +
-		"var hash = \"\"; if(window.location.hash) {hash = window.location.hash.substring(1);} window.location.replace(\"%s\" + hash)\n" +
+		"var hash = \"\"; if(window.location.hash) {hash = \"/#\" + window.location.hash;} window.location.replace(\"%s\" + hash)\n" +
```

Comment:
May as well include the bang too (to avoid another redirect).
hash = \"/#!\" + window.location.hash;